Debulleting the Porch - Guides pages (#557)
nephio-project/porch: docs/content/en/docs/4_tutorials_and_how-tos/working_with_package_revisions/_index.md

**PackageRevision Operations:**

| Operation | Git Command | Description |
|-----------|---------|-------------|
| Creation | `init`, `clone`, `copy` | Create new PackageRevisions from scratch, existing packages, or new revisions |
| Inspection | `get` | List and view PackageRevision information and metadata |
| Content Management | `pull`, `push` | Move PackageRevision content between Git repositories and local filesystem |
| Lifecycle Management | `propose`, `approve`, `reject` | Control PackageRevision workflow states |
| Upgrading | `upgrade` | Create new revision upgrading downstream to more recent upstream package |

> | Deletion | `propose-delete`, `del` | Propose deletion of published PackageRevisions, then delete them |

Reviewer: nice table approach

Reviewer 2: Formatted it to look even nicer in the source markdown :)

- **Proposed**: Ready for review, contents are immutable. To make further changes, reject back to Draft. Revision number remains 0.
- **Published**: Approved and immutable. Revision number increments to 1+.

> PackageRevisions follow the following lifecycle transitions:

Reviewer: "follow the following" seems like odd wording. maybe use something like
```suggestion
PackageRevisions use the following lifecycle transitions:
```

1. **Draft → Proposed**: `porchctl rpkg propose` - Signal readiness for review
2. **Proposed → Published**: `porchctl rpkg approve` - Approve and make immutable
3. **Proposed → Draft**: `porchctl rpkg reject` - Return for more work

> An additional state is **DeletionProposed**. In his state, the PackageRevision is marked for deletion, pending approval.

Reviewer: i think this state should be with the others and not separated in another point below

---

## PackageRevision Naming

Porch generates PackageRevision names automatically using the `{repositoryName}.{packageName}.{workspaceName}` format. An example PackageRevision naming would be `porch-test.my-first-package.v1`.

The name consists of the following components:

- **Repository Name**: Name of the registered Git repository
- **Package Name**: Logical name for the package (can have multiple revisions)
- **Workspace Name**: Unique identifier within the package (maps to Git branch/tag)

{{% alert title="Note" color="primary" %}}

> Workspace names must be unique within a package. Multiple PackageRevisions can share the same package name with different workspaces and published PackageRevisions get tagged in Git using the workspace name 

Reviewer 2: ```suggestion
Workspace names must be unique within a package. Multiple PackageRevisions can share the same package name with different workspaces and published PackageRevisions get tagged in Git using the workspace name.
```

Author: corrected

- **Proposed**: Stored in `proposed/{workspace}` branch  
- **Published**: Tagged as `{workspace}` and stored in main branch

> There are two repository types: Blueprint Repositories and Deployment Repositories. Blueprint Repositories contain upstream package templates for cloning, while Deployment Repositories store deployment-ready packages. These are marked with the `--deployment` flag.

Reviewer 2: Should we really capitalize "Blueprint Repositories" and "Deployment Repositories"?

I'm thinking something like:

```suggestion
Porch differentiates two types of repositories: "blueprint" and "deployment". Blueprint repositories are meant to contain upstream package templates for cloning, while deployment repositories should store deployment-ready packages. The latter are marked with the `--deployment` flag.
```

**PackageRevision stuck in Draft?**

- Check readiness conditions with the `porchctl rpkg get <PACKAGE-REVISION> -o yaml | grep -A 5 conditions` command
- Verify that all required fields are populated in the Kptfile
- Check for pipeline function errors in the Porch server logs

**Push fails with conflict?**

> - Pull the latest version first with the `porchctl rpkg pull <PACKAGE-REVISION> ./dir` command

Reviewer 2: ```suggestion
- Pull the latest version first with the `porchctl rpkg pull <PACKAGE-REVISION> <directory>` command
```
